lib/phoenix_live_view/static.ex

```diff
@@ -187,6 +187,8 @@ defmodule Phoenix.LiveView.Static do
       Utils.configure_socket(
         %Socket{
           id: to_string(child_id),
+          root_view: parent.root_view,
+          view: view,
           endpoint: endpoint,
           root_pid: parent.root_pid,
           parent_pid: self()
```

Comment:
Looking at this code, the child LiveView is not getting a router, which means any attempt to do a live patch/redirect  in a disconnected router will fail by saying the view is not mounted at the root. This is ok, because we at least fail, but slightly misleading because you now can redirect from a child LiveView, just not on mount/handle_params.

And I think this points out to the following: we can actually redirect from handle_params, as long as the view is connected. So I think we can simply this a bit by removing the @valid_child_redirect check from the LiveView Channel and instead check if we are in a disconnected child render whenever we call push_redirect/patch. Checking it there will also allow us to raise a proper error message.

I will try to push something along these lines.

Reply:
I have pushed a fix. :)